src/Engine/ProtoAssociative/CodeGen.cs

```diff
@@ -3734,6 +3810,124 @@ private AssociativeNode DFSEmitSplitAssign_AST(AssociativeNode node, ref List<As
             return node;
         }
 
+        /// <summary>
+        /// Converts lhs ident lists to a function call
+        /// a.x = 10 
+        ///     -> t = a.%set_x(10)
+        ///     
+        /// a.x.y = b + c 
+        ///     -> a.x.%set_y(b + c)
+        ///     
+        /// a.x[0] = 10
+        ///     ->  tval = a.%get_x()
+        ///         tval[0] = 10         
+        ///         tmp = a.%set_x(tval)
```

Comment:
The main reason for conveting the lhs identlist to a setter is to have a unified way of treating  identlists. 
We already do this at codegen in here: EmitGetterSetterForIdentList. I moved that part much earlier - before SSA transform into the function: TransformLHSIdentList

I added a task to remove EmitGetterSetterForIdentList from the codegen:
http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7433